html/semantics/forms/form-submission-0/multipart-formdata.window.js

```diff
@@ -0,0 +1,444 @@
+form({
+  name: "a",
+  value: new File([], "b\rc", { type: "text/plain" }),
+  expected: {
+    name: "a",
+    filename: "b%0Dc",
+    value: "",
+  },
+  description: "\\r in filename",
+});
+
+form({
+  name: "a",
+  value: new File([], "b\r\nc", { type: "text/plain" }),
+  expected: {
+    name: "a",
+    filename: "b%0D%0Ac",
+    value: "",
+  },
+  description: "\\r\\n in filename",
+});
+
+form({
+  name: "a",
+  value: new File([], "b\n\rc", { type: "text/plain" }),
+  expected: {
+    name: "a",
+    filename: "b%0A%0Dc",
+    value: "",
+  },
+  description: "\\n\\r in filename",
+});
+
+form({
+  name: 'a"b',
+  value: "c",
+  expected: {
+    name: "a%22b",
+    value: "c",
+  },
+  description: "double quote in name",
+});
+
+form({
+  name: "a",
+  value: 'b"c',
+  expected: {
+    name: "a",
+    value: 'b"c',
+  },
+  description: "double quote in value",
+});
+
+form({
+  name: "a",
+  value: new File([], 'b"c', { type: "text/plain" }),
+  expected: {
+    name: "a",
+    filename: "b%22c",
+    value: "",
+  },
+  description: "double quote in filename",
+});
+
+form({
+  name: "a'b",
+  value: "c",
+  expected: {
+    name: "a'b",
+    value: "c",
+  },
+  description: "single quote in name",
+});
+
+form({
+  name: "a",
+  value: "b'c",
+  expected: {
+    name: "a",
+    value: "b'c",
+  },
+  description: "single quote in value",
+});
+
+form({
+  name: "a",
+  value: new File([], "b'c", { type: "text/plain" }),
+  expected: {
+    name: "a",
+    filename: "b'c",
+    value: "",
+  },
+  description: "single quote in filename",
+});
+
+form({
+  name: "a\\b",
+  value: "c",
+  expected: {
+    name: "a\\b",
+    value: "c",
+  },
+  description: "backslash in name",
+});
+
+form({
+  name: "a",
+  value: "b\\c",
+  expected: {
+    name: "a",
+    value: "b\\c",
+  },
+  description: "backslash in value",
+});
+
+form({
+  name: "a",
+  value: new File([], "b\\c", { type: "text/plain" }),
+  expected: {
+    name: "a",
+    filename: "b\\c",
+    value: "",
+  },
+  description: "backslash in filename",
+});
+
+form({
+  name: "áb",
+  value: "ç",
+  expected: {
+    name: "\xC3\xA1b",
+    value: "\xC3\xA7",
+  },
+  description: "non-ASCII in name and value",
+});
+
+form({
+  name: "a",
+  value: new File([], "ə.txt", { type: "text/plain" }),
+  expected: {
+    name: "a",
+    filename: "\xC9\x99.txt",
+    value: "",
+  },
+  description: "non-ASCII in filename",
+});
+
+form({
+  name: "aəb",
+  value: "c\uFFFDd",
+  formEncoding: "windows-1252",
+  expected: {
+    name: "a&#601;b",
+    value: "c&#65533;d",
+  },
+  description: "characters not in encoding in name and value",
+});
+
+form({
+  name: "á",
+  value: new File([], "💩", { type: "text/plain" }),
+  formEncoding: "windows-1252",
+  expected: {
+    name: "\xE1",
+    filename: "&#128169;",
+    value: "",
+  },
+  description: "character not in encoding in filename",
+});
+
+function form({ name, value, expected, formEncoding = "utf-8", description }) {
```

Comment:
This function looks to be sharable with text-plain.window.js and urlencoded2.window.js. Please make a helper js file for this function, and refer to it in these tests by "// META: script=".

Reply:
Done. Other than just the code duplication in the `form` function being repeated with minor changes across the three test files, the contents of the `form` function itself were duplicated for each of the runs of each test (what I'm calling "normal form" vs "`formdata` event"), so I also moved that code to a separate function.